crates/data_components/src/duckdb.rs

```diff
@@ -207,7 +241,7 @@ impl TableProviderFactory for DuckDBTableProviderFactory {
             Some(Engine::DuckDB),
         ));
 
-        let read_write_provider = DuckDBTableWriter::create(read_provider, duckdb);
+        let read_write_provider = DuckDBTableWriter::create(read_provider, duckdb, on_conflict);
```

Comment:
We pass the `OnConflict` behavior to the TableWriter struct, which keeps it around and eventually passes it to the main DuckDB struct when inserting. This pattern is the same for Postgres/Sqlite.